src/controller/SetUpCodePairer.cpp

```diff
@@ -312,12 +312,17 @@ void SetUpCodePairer::NotifyCommissionableDeviceDiscovered(const Dnssd::Discover
 
     ChipLogProgress(Controller, "Discovered device to be commissioned over DNS-SD");
 
-    Inet::InterfaceId interfaceId =
-        nodeData.resolutionData.ipAddress[0].IsIPv6LinkLocal() ? nodeData.resolutionData.interfaceId : Inet::InterfaceId::Null();
-    Transport::PeerAddress peerAddress =
-        Transport::PeerAddress::UDP(nodeData.resolutionData.ipAddress[0], nodeData.resolutionData.port, interfaceId);
-    mDiscoveredParameters.emplace();
-    mDiscoveredParameters.back().SetPeerAddress(peerAddress);
+    auto & resolutionData = nodeData.resolutionData;
+    for (size_t i = 0; i < resolutionData.numIPs; i++)
+    {
+        Inet::InterfaceId interfaceId =
+            resolutionData.ipAddress[i].IsIPv6LinkLocal() ? resolutionData.interfaceId : Inet::InterfaceId::Null();
+        Transport::PeerAddress peerAddress =
+            Transport::PeerAddress::UDP(resolutionData.ipAddress[i], resolutionData.port, interfaceId);
+        mDiscoveredParameters.emplace();
+        mDiscoveredParameters.back().SetPeerAddress(peerAddress);
```

Comment:
What I think we should do is:
1. Switch `mDiscoveredParameters` from `std::queue` to `std::deque`.
2. Switch the BLE code to `emplace_front` into the queue.
3. Have this code `emplace_back` into the queue.
4. Have the code trying the next option `pop_front`.

That way, if we get a bunch of IPs here that will not starve the one BLE thing we find, if we find one.